src/parser/GQLParser.h

```diff
@@ -16,6 +16,21 @@ namespace nebula {
 class GQLParser {
 public:
     GQLParser() : parser_(scanner_, error_, &sentences_) {
+        // Callback invoked by GraphScanner
+        auto readBuffer = [this] (char *buf, int maxSize) -> int {
+            // Reach the end
+            if (pos_ >= end_) {
+                pos_ = nullptr;
+                end_ = nullptr;
+                return 0;
+            }
+            int left = end_ - pos_;
+            auto n = maxSize > left ? left : maxSize;
+            ::memcpy(buf, pos_, n);
+            pos_ += n;
+            return n;   // Number of bytes we actually filled in `buf'
+        };
+        scanner_.setReadBuffer(std::move(readBuffer));
```

Comment:
Great work. let me benefit greatly by this PR. I have a little question ：
Here, we using memcpy, copy the memory from `pos_` to `but`.  How to destroy both?
Is it the below destructor? 
```
yyFlexLexer::~yyFlexLexer()
{
	delete [] yy_state_buf;
	yyfree(yy_start_stack  );
	yy_delete_buffer( YY_CURRENT_BUFFER );
	yyfree(yy_buffer_stack  );
}
```

Reply:
Yeah. Flex itself maintains a mutable buffer, of which we don't bother to take care of the lifetime.

But you reminded me to find an existing leak that happens on the failure case. Thanks for that.

Reply:
Addressed, please have a check.